contracts/strategies/stkCvxFXS/Strategy.sol

```diff
@@ -0,0 +1,133 @@
+import "../../../interfaces/ICvxFxsStaking.sol";
+import "../../../interfaces/IHarvester.sol";
+
+contract stkCvxFxsStrategy is Ownable {
+    using SafeERC20 for IERC20;
+    address public constant CVXFXS_TOKEN =
+        0xFEEf77d3f69374f66429C91d732A244f074bdf74;
+
+    address public immutable vault;
+    ICvxFxsStaking constant cvxFxsStaking =
+        ICvxFxsStaking(0x49b4d1dF40442f0C31b1BbAEA3EDE7c38e37E31a);
+
+    address public harvester;
+    uint256 public constant FEE_DENOMINATOR = 10000;
+
+    constructor(address _vault) {
+        vault = _vault;
+    }
+
+    /// @notice Set approvals for the contracts used when swapping & staking
+    function setApprovals() external {
+        IERC20(CVXFXS_TOKEN).safeApprove(address(cvxFxsStaking), 0);
+        IERC20(CVXFXS_TOKEN).safeApprove(
+            address(cvxFxsStaking),
+            type(uint256).max
+        );
+    }
+
+    /// @notice Update the harvester contract
+    /// @param _harvester address of the new contract
+    function setHarvester(address _harvester) external onlyOwner {
+        require(_harvester != address(0));
+        harvester = _harvester;
+        // ensures all rewards are redirected to harvester
+        // if regular claim reward is triggered on staking contract
+        cvxFxsStaking.setRewardRedirect(harvester);
+    }
+
+    /// @notice Query the amount currently staked
+    /// @return total - the total amount of tokens staked
+    function totalUnderlying() public view returns (uint256 total) {
+        return cvxFxsStaking.balanceOf(address(this));
+    }
+
+    /// @notice Deposits all underlying tokens in the staking contract
+    function stake(uint256 _amount) external onlyVault {
+        cvxFxsStaking.stake(_amount);
+    }
+
+    /// @notice Withdraw a certain amount from the staking contract
+    /// @param _amount - the amount to withdraw
+    /// @dev Can only be called by the vault
+    function withdraw(uint256 _amount) external onlyVault {
+        cvxFxsStaking.withdraw(_amount);
+        IERC20(CVXFXS_TOKEN).safeTransfer(vault, _amount);
+    }
+
+    /// @notice Claim rewards and swaps them to cvxFXS for restaking
+    /// @dev Can be called by the vault only
+    /// @param _caller - the address calling the harvest on the vault
+    /// @param _minAmountOut - min amount of cvxFxs expected
+    /// @return harvested - the amount harvested
+    function harvest(address _caller, uint256 _minAmountOut)
+        external
+        onlyVault
+        returns (uint256 harvested)
+    {
+        // claim rewards
+        cvxFxsStaking.getReward(address(this), harvester);
+
+        uint256 _cvxFxsBalance = IHarvester(harvester).processRewards();
+        require(_cvxFxsBalance >= _minAmountOut, "slippage");
+
+        uint256 _stakingAmount = _cvxFxsBalance;
+
+        if (_cvxFxsBalance > 0) {
+            // if this is the last call, no fees
+            if (IGenericVault(vault).totalSupply() != 0) {
```

Comment:
`IGenericVault(vault)` can be assigned to a variable to both improve readability and avoid redundant storage reads.

Reply:
Same with `IERC20(CVXFXS_TOKEN)`.